web/search.dart

```diff
@@ -203,24 +210,54 @@ void initializeSearch(
         query, "<strong class='tt-highlight'>$sanitizedText</strong>");
   }
 
+  var categoriesMap = <String, Element>{};
+
+  // Maps a suggestions library/class to the other suggestions that search found if any.
+  void categorize(Element suggestionLibrary, Element suggestion) {
+    var input = suggestionLibrary.innerHtml;
+
+    if (input == null) {
+      return;
+    }
+
+    var element = categoriesMap[input];
+    if (element != null) {
+      element.append(suggestion);
+      categoriesMap[input] = element;
```

Comment:
This line should not be needed. When you extract `categoriesMap[input]` to a local variable, `element`, it does not make a copy of the object. So `element.append` above will apply to the object in the Map.